src/src/com/microsoft/aad/adal/DateTimeAdapter.java

```diff
@@ -50,12 +50,31 @@
             DateFormat.DEFAULT);
 
     private final DateFormat iso8601Format = buildIso8601Format();
+    private final DateFormat enUs24HourFormat = buildEnUs24HourDateFormat();
+    private final DateFormat local24HourFormat = buildLocal24HourDateFormat();
 
     private static DateFormat buildIso8601Format() {
         DateFormat iso8601Format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US);
         iso8601Format.setTimeZone(TimeZone.getTimeZone("UTC"));
         return iso8601Format;
     }
+    
+    /**
+     * Add new en-us date format for parsing date string if it doesn't contain AM/PM.
+     */
+    private static DateFormat buildEnUs24HourDateFormat() {
+        DateFormat newformat = new SimpleDateFormat("MMM dd, yyyy HH:mm:ss", Locale.US);
```

Comment:
server return both expires in and expires on(will be milliseconds since January 1, 1970, 00:00:00 GMT)